### PR TITLE
Clarify language for builds with subscriptions

### PR DESCRIPTION
**Pull request by adambkaplan**
- Replace "entitled" to "subscription" where possible
- Other nits

/assign @bmcelvee 

/cc @bparees

### CODE REVIEW

File: modules/builds-create-imagestreamtag.adoc [hunk 1]
```diff
@@ -3,10 +3,12 @@
...
 [id="builds-create-imagestreamtag_{context}"]
-= Creating an ImageStreamTag to the Red Hat Container Catalog
+= Creating an ImageStreamTag to the Red Hat Universal Base Image
 
-To run entitled builds, you must create an `ImageStream` to reference the
-universal base image (UBI).
+To use Red Hat subscriptions within a build, you should create an `ImageStream`
+to reference the universal base image (UBI).
+
+Builds which reference the UBI directly from registry.redhat.io will require a pull secret.
```
Comment (bmcelvee) [inline modules/builds-create-imagestreamtag.adoc]: s/which/that

File: modules/builds-source-secrets-entitlements.adoc [hunk 2]
```diff
@@ -3,20 +3,20 @@
...
 
-Builds that use entitlements to install subscription content must include the
+Builds that use Red Hat subscriptions to install content must include the
 entitlement keys as a build secret.
 
 .Prerequisites
 
-You must have access to Red Hat entitlements, and the entitlements must have
-separate public and private key files.
+You must have access to Red Hat entitlements (through your subscription), and
```
Comment (bmcelvee) [inline modules/builds-source-secrets-entitlements.adoc]: no parentheses needed

**Comment by bmcelvee** [comment]
Thanks again, Adam! I made a few notes, but am also happy to make the changes in a follow-up PR.

**Comment by adambkaplan** [comment]
@bmcelvee fixed

**Comment by bmcelvee** [comment]
No QE review required.

**Pull request merged**